readchar/readchar_linux.py

```diff
@@ -1,18 +1,78 @@
 # -*- coding: utf-8 -*-
-# Initially taken from:
-# http://code.activestate.com/recipes/134892/
-# Thanks to Danny Yoo
+# Copyright (c) 2014 - 2017 Miguel Ángel García (@magmax9).
+# Licensed under the MIT license.
+# This code based on code snippet from the python 2 docs:
+# https://docs.python.org/2/faq/library.html
+#     #how-do-i-get-a-single-keypress-at-a-time
+
+
 import sys
-import tty
+import os
 import termios
+import fcntl
+from . import key
+
+
+def get_char(test_stream=None):
+    charbuffer = ''
+    readstream = read_char
+
+    if test_stream:
+        def altreader(*args):
+            return test_stream()
+        readstream = altreader
```

Comment:
Dislike. It is difficult to understand.
Why not just:
```
readstream = test_stream or read_char
```

This is the same than lines 18-23.